base/c.jl

```diff
@@ -143,8 +143,8 @@ function transcode end
 transcode{T<:Union{UInt8,UInt16}}(::Type{T}, src::Vector{T}) = src
 transcode(::Type{Int32}, src::Vector{UInt32}) = reinterpret(Int32, src)
 
-function transcode(::Type{UInt16}, src::Vector{UInt8})
-    dst = UInt16[]
+function transcode{T<:Union{UInt16, Int32}}(Dst_et::Type{T}, src::Vector{UInt8})
+    dst = Dst_et[]
     i, n = 1, length(src)
```

Comment:
I believe you can just write

```julia
function transcode{T<:Union{UInt16, Int32}}(::Type{T}, src::Vector{UInt8})
    dst = T[]
    ...
end
```

and use `T` instead of `Dst_et` which would look cleaner IMO.